src/alpaka/test/alpaka/AtomicPairCounter_t.cc

```diff
@@ -2,24 +2,19 @@
 #include <iostream>
 
 #include "AlpakaCore/alpakaConfig.h"
+#include "AlpakaCore/alpakaWorkDivHelper.h"
 #include "AlpakaCore/AtomicPairCounter.h"
 
 using namespace ALPAKA_ACCELERATOR_NAMESPACE;
 
 struct update {
   template <typename T_Acc>
   ALPAKA_FN_ACC void operator()(
-      const T_Acc &acc, cms::Alpaka::AtomicPairCounter *dc, uint32_t *ind, uint32_t *cont, uint32_t n) const {
-    // Global thread index in grid
-    const uint32_t threadIdxGlobal(alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u]);
-    const uint32_t threadDimension(alpaka::workdiv::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u]);
+      const T_Acc &acc, cms::alpakatools::AtomicPairCounter *dc, uint32_t *ind, uint32_t *cont, uint32_t n) const {
+    const auto &[firstElementIdxGlobal, endElementIdxGlobal] =
+        cms::alpakatools::element_global_index_range(acc, Vec1::all(n));
 
-    // Global element index (obviously relevant for CPU only, for GPU, i = threadIndexGlobal only)
-    const uint32_t firstElementIdxGlobal = threadIdxGlobal * threadDimension;
-    const uint32_t endElementIdxGlobalUncut = firstElementIdxGlobal + threadDimension;
-    const uint32_t endElementIdxGlobal = std::min(endElementIdxGlobalUncut, n);
-
-    for (uint32_t i = firstElementIdxGlobal; i < endElementIdxGlobal; ++i) {
+    for (uint32_t i = firstElementIdxGlobal[0u]; i < endElementIdxGlobal[0u]; ++i) {
```

Comment:
>Another, related question that comes to my mind now is the use of unsigned int vs int in loop variables. 

Yes true. Maybe using `alpaka_common::Idx`? Since it is the type of what we have inside the `Vec` anyway.

Reply:
I was thinking more of the definition of `alpaka_common::Idx` itself. Here `auto` would do the job. Not super-important anyway for the prototyping itself.

Reply:
>I was thinking more of the definition of alpaka_common::Idx itself. 

Ha yes would need to think about that